src/main/java/com/openshift/internal/restclient/model/ImageStream.java

```diff
@@ -41,4 +48,45 @@ public DockerImageURI getDockerImageRepository() {
 		return new DockerImageURI(asString(IMAGESTREAM_DOCKER_IMAGE_REPO));
 	}
 
+	@Override
+	public void setTag(String newTag, String fromTag) {
+		ModelNode tags = get(IMAGESTREAM_SPEC_TAGS);
+		ModelNode tag = new ModelNode();
+		tag.get("name").set(newTag);
+		ModelNode from = new ModelNode();
+		from.get("kind").set("ImageStreamTag");
+		from.get("name").set(fromTag);
+		tag.get("from").set(from);
+		tags.add(tag);
+	}
+
+	@Override
+	public String getImageId(String tagName) {
+		String imageId = null;
+		ModelNode tags = get(IMAGESTREAM_STATUS_TAGS);
+		if (tags == null || tags.getType() != ModelType.LIST)
+			return null;
+		List<ModelNode> tagWrappers = tags.asList();
+		if (tagWrappers != null) {
+			for (ModelNode tagWrapper : tagWrappers) {
+				ModelNode tag = tagWrapper.get(TAG);
+				ModelNode items = tagWrapper.get(ITEMS);
+				if (tag != null && tag.asString().equals(tagName) && items != null) {
+					List<ModelNode> itemWrappers = items.asList();
+					for (ModelNode itemWrapper : itemWrappers) {
```

Comment:
this could be extracted to a separate method #getImageId(List<ModelNode> itemWrappers) for added code clarity/brevity